scalajvm/app/helpers/ActionWithTx.scala

```diff
@@ -7,12 +7,16 @@ import scala.concurrent.Future
 
 class RequestWithSession[A](request: Request[A])(implicit val dbSession: DBSession) extends WrappedRequest[A](request)
 
-object ActionWithTx extends ActionBuilder[RequestWithSession] {
+class ActionWithTx(cc: ControllerComponents) extends ActionBuilder[RequestWithSession, AnyContent] {
```

Comment:
We now have to instantiate `ActionWithTx` all over the place, which is ugly. Furthermore, I couldn't figure out how to pass `cc` implicitly: it's either not picked up, or Scala doesn't like my syntax. Injecting the dependency didn't work for me either. I'm probably just doing something wrong; can someone please advise me on how to get rid of `new` and explicit `cc` at all callsites?

Reply:
`class ActionWithTx()(implicit cc: ControllerComponents)` ?

Reply:
Doesn't quite work, and doesn't quite solve the problem:

```diff
diff --git a/scalajvm/app/cors/CorsController.scala b/scalajvm/app/cors/CorsController.scala
index e8ad738..c59367d 100644
--- a/scalajvm/app/cors/CorsController.scala
+++ b/scalajvm/app/cors/CorsController.scala
@@ -26,10 +26,11 @@ class CorsController @Inject()(cc: ControllerComponents, configuration: Configur
     }
   }
 
-  protected def corsActionWithTx(responseAction: RequestWithSession[AnyContent] => Result)(implicit cc: ControllerComponents) =
-    (new ActionWithTx(cc)) { request =>
+  protected def corsActionWithTx(responseAction: RequestWithSession[AnyContent] => Result)(implicit cc: ControllerComponents) = {
+    (new ActionWithTx) { request =>
       val response = responseAction(request)
       val headers = corsHeaders(request.headers)
       response.withHeaders(headers: _*)
     }
+  }
 }
diff --git a/scalajvm/app/helpers/ActionWithTx.scala b/scalajvm/app/helpers/ActionWithTx.scala
index c10b49a..bc25945 100644
--- a/scalajvm/app/helpers/ActionWithTx.scala
+++ b/scalajvm/app/helpers/ActionWithTx.scala
@@ -7,7 +7,7 @@ import scala.concurrent.Future
 
 class RequestWithSession[A](request: Request[A])(implicit val dbSession: DBSession) extends WrappedRequest[A](request)
 
-class ActionWithTx(cc: ControllerComponents) extends ActionBuilder[RequestWithSession, AnyContent] {
+class ActionWithTx()(implicit cc: ControllerComponents) extends ActionBuilder[RequestWithSession, AnyContent] {
   override def invokeBlock[A](request: Request[A], block: (RequestWithSession[A]) => Future[Result]): Future[Result] = {
     DB localTx { implicit session =>
       block(new RequestWithSession[A](request))
```

leads to very confusing errors in blocks (that's what I called "Scala doesn't like my syntax"):

```
$ sbt compile
--->8--- snip --->8---
[error] /home/minoru/src/loglist/scalajvm/app/cors/CorsController.scala:30:26: missing parameter type
[error]     (new ActionWithTx) { request =>
[error]                          ^
--->8--- snip --->8---
```

I can bind the newly created object object to a `val`, then it'll compile fine, but I don't want to introduce a `val` at every callsite.

Furthermore, this still requires `new` at each callsite. I was hoping there's something I can do to regain the ease with which it was done before: just `ActionWithTx { request => /* do stuff */ }`.

Reply:
Whatever, it will be fine.